cupy/core/core.pyx

```diff
@@ -4255,3 +4312,120 @@ cpdef ndarray scan(ndarray a, ndarray out=None):
                  block=(2 * block_size - 1,),
                  args=(out,))
     return out
+
+
+# -----------------------------------------------------------------------------
+# partition
+# -----------------------------------------------------------------------------
+
+@util.memoize(for_each_device=True)
```

Comment:
Cloud you create another file? For example `core/_partition.py`?

Reply:
The function depends on `carray.pxi` and it cannot work outside of `core.pyx`. We need to change `carray.pxi` to split the function from `core.pyx`.

Reply:
OK